pkg/cluster/cluster.go

```diff
@@ -211,11 +211,10 @@ func (c *Cluster) run(stopC <-chan struct{}, wg *sync.WaitGroup) {
 		if needDeleteCluster {
 			c.logger.Infof("deleting cluster")
 			c.delete()
+			go c.reportFailedStatus()
```

Comment:
> Reconcile only fail when it should. Create an new issue and let's discuss elsewhere.

This pr is incorrect. Now the code logic is that we report TPR status to failure when the cluster TPR is deleted. This makes no sense to me.